src/js/modules/media/selectors.js

```diff
@@ -0,0 +1,65 @@
+import { createSelector } from 'reselect';
+import { getArticles } from "../articles/selectors";
+import { objectFilter } from "../../utils";
+import { getUsers} from "../users/selectors";
+
+Object.filter = objectFilter;
+
+const getMedia = state => state.media.media;
+
+export const matchMediaToArticles = createSelector(
+  [ getMedia, getArticles ],
+  ( medias, articles ) => {
+    const newArticleObject = {};
+    Object.keys(articles).map((article) => {
```

Comment:
Also, you aren't mapping by an article. You are mapping by the articleSlug. So articleSLug should be the parameter.